### ISSUE TITLE
Laravel default locale is not in the supportedLocales array.

### ISSUE THREAD

**Issue by nWidart** [mcamara/laravel-localization#198]
Hello,

Some users are getting the following error:

```
UnsupportedLocaleException in LaravelLocalization.php line 119:
Laravel default locale is not in the supportedLocales array.
```

I can reproduce this by hardcoding the `App::setLocale('fr')`, for instance.
It is true dat in the `laravellocalization.php` config file, I only have `nl` and `en` enabled. (Though there is one guy which is on a 'nl' os, and is also getting that error.)

Though the `app.locale` config value is set to `en`, same as the `fallback_locale`. Why isn't it using that locale if it can't find a locale ?

Basically I'd like that if the user comes from a locale that isn't present in `laravellocalization.php` config file, take the `default_locale`, from `app.php`..

Thanks,

**Comment by nWidart** [comment]
Turns out I was using the following code to set the locale:

``` php

$lang = Request::getPreferredLanguage(['fr', 'en', 'de', 'nl']);

if (App::environment() == 'testing') {
    $lang = 'fr';
}

App::setLocale($lang);
```

But which was overwriting the app.`locale key`. 

I now just use `LaravelLocalization::setLocale();` which seems to have fixed it.

**Comment by dsampaolo** [comment]
I have the same error today on a fresh Laravel 5.2 install. My default locale is set to 'en' and this language is enabled in config/laravellocalization.php

For some reason, $this->configRepository->get('app.locale') (in LaravelLocalization.php, line 114) returns null.

**Comment by xitude** [comment]
I too have this issue, been pulling my hair out trying to get past it.

**Comment by floodedcodeboy** [comment]
So I have this issue too. 
I'm using a vagrant box (via phansible.com, it's quick and easy, ubuntu 14.04) 
any chance that the server needs locales installed?!? 
i actually installed ALL possible locales and  briefly got rid of the error message. 
please for the love of >insert deity here<. help.

**Comment by xitude** [comment]
I opened a new issue #341 could you help detail it out with stack traces and any more information you have?

**Comment by gitsakil09** [comment]
I think I have found a solution.I am not 100% sure but it works perfectly for me. :) :) you can check it. As I said I am not a pro so just give a try.Could help you.
just replace line number 114 in laravellocalization.php as follow

$this->defaultLocale = $this->configRepository->get('app.locale'); 
line is replaced by 
$this->defaultLocale = $this->configRepository->get('app.fallback_locale');

Tnks

**Comment by diegodss** [comment]
I had the same error, but for me it was a silly mistake: I added the some code on route.php

`    // app/Http/routes.php

    Route::group(['prefix' => LaravelLocalization::setLocale()], function()`

Before run the publish command:

`php artisan vendor:publish --provider="Mcamara\LaravelLocalization\LaravelLocalizationServiceProvider"`

I obviously needed to first publish the plugin, config the supportedLocales, and then add any code that I want.

**Comment by spygibas** [comment]
Test to comment  throw error line and run site test swap language 1 time and uncomment
//throw new UnsupportedLocaleException('Laravel default locale is not in the supportedLocales array.');

after change language

and uncomment again

**Comment by webbhelp** [comment]
For me it was because the value in config/app.php | 'locale' was set to a language that was no activated.

**Comment by azamat96** [comment]
I'm enable unsupported locales from this file <YOUR_PROJECT_NAME>/vendor/mcamara/laravel-localization/src/config/config.php

**Comment by rrudowski** [comment]
azamat96 great solution Thanks!

**Comment by pammel** [comment]
Same problem here. Manually clearing the config cache solved my problem:
rm bootstrap/cache/config.php
